js/ai/src/document.ts

```diff
@@ -40,9 +41,12 @@ export type MediaPart = z.infer<typeof MediaPartSchema>;
 export const PartSchema = z.union([TextPartSchema, MediaPartSchema]);
 export type Part = z.infer<typeof PartSchema>;
 
+// We need both metadata and embedMetadata because they can
+// contain the same fields (e.g. video start/stop) with different values.
 export const DocumentDataSchema = z.object({
   content: z.array(PartSchema),
   metadata: z.record(z.string(), z.any()).optional(),
+  embedMetadata: z.record(z.string(), z.unknown()).optional(),
```

Comment:
please use the metadata field, separate `embedMetadata` field is not necessary.

Reply:
Yes it is. As I stated in the comments above: 
    // We need both metadata and embedMetadata because they can
    // contain the same fields (e.g. video start/stop) with different values.
We would not have needed this if we went with one document per embeddings because they would have matched, but you wanted multiple. So now we have extra embed data that is different from the request metadata but has the same fields, so they can't be stored in metadata. 
e.g. 
metadata: {
someInfo: "blah",
startEmbeddingSeconds: 0,
endEmbeddingSeconds: 120,
},
embedMetadata: {
startEmbeddingSeconds: 30,
endEmbeddingSeconds: 60,
}

Reply:
no, you would do it like this:

```ts
const doc = Document.fromMedia('gs://.../', 'video/mp4', {
  someInfo: "blah",
  startEmbeddingSeconds: 0,
  endEmbeddingSeconds: 120,
});
const embedding = ai.embed({
  embedder: multimodalEmbedding001,
  content: doc,
  metadata: {
    startEmbeddingSeconds: 30,
    endEmbeddingSeconds: 60,
  }
});
```

document always represents the original document.

Reply:
Right, and then when you retrieve it from storage, you have NO IDEA which embedding it used or which part of the video to send except "the whole big chunk". So there would be no point in getting any additional embeddings past a single one for the document. We had this discussion...

Reply:
Perhaps it's not clear - the "embedMetadata that gets added to the original document is what comes back from the embedder in the embedding response metadata."

Reply:
So if you specify a video document from 0-120 seconds and you get super fine embeddings every 8 seconds, then you get a lot of embeddings back. But that's no good to you if they all point to the same document. Vector storage stores by embedding (number array) only, not embedding plus metadata, so if you want to know which part of the video a particular embedding is for, then you need to store the embedding metadata somewhere and the only option is in the document. It's messy, but you wanted support for multiple embeddings returned per incoming document. And documents are not unique unless they also contain the embedMetadata. There's no point in having 5 embeddings all pointing to the same document.

Reply:
Yes, but then Embedding -> Storage -> Retriever -> Document.
And now where is the embedding metadata? If you want to use it, then it needs to be in the document. Given a document with only the original metadata you have no way of knowing which part of the video to send along in your RAG query except the whole big chunk which negates the usefulness of fine grained chunking.

Reply:
Also, for most of our vector stores, the id that we store the information under is the MD5 hash of the JSOn.stringify(document). So if 5 embeddings point to the exact same document, then we would only store either the first or the last depending on if we overwrite or skip upon finding an entry already in the store. We could change how the id is calculated, but I think that's another breaking change... which seemed unnecessary.

Reply:
It just occurred to me that when calling `ai.embed` things like video chunking or truncation are not "metadata", but options:

```ts
const embedding = ai.embed({
  embedder: multimodalEmbedding001,
  content: Document.fromMedia('gs://.../', 'video/mp4'),
  options: {
    videoStartEmbeddingSeconds: 30,
    videoEndEmbeddingSeconds: 60,
  }
});
```

metadata should not be used for action parameters.

Reply:
it's OK to put embedMetadata into metadata if really necessary, but it's all just metadata, it does not require a separate field.

```
metadata: {
  embedMetadata: {
      startEmbeddingSeconds: 30,
      endEmbeddingSeconds: 60,
  }
}
```

Reply:
done